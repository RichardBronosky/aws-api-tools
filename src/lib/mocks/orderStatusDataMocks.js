export const orderStatus = {
  AbleToReschedule: false,
  AdditionalDirections: '**** ALL DAY DELIVERY**** ****NO TIME FRAMES****',
  AffiliateData: '',
  Balance: '0.00',
  BillingAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    State: 'GA',
    Zip: '30328',
  },
  Customer: {
    AllowText2AltPhoneNumber: false,
    AllowText2PhoneNumber: false,
    AltPhoneNumber: '',
    CMKey: '01*202/292-8491',
    CustomerID: 'SE006284958',
    Division: 'SE',
    FirstName: 'SARAH',
    LastName: 'SMITH',
    PhoneNumber: '202/292-8491',
  },
  DeliveryAddress: {
    Address1: '1112 MARSH TRAIL CIR',
    Address2: '',
    City: 'ATLANTA',
    Plus4: '',
    State: 'GA',
    Zip: '30328',
  },
  DeliveryCharge: '119.99',
  DeliveryDate: '2019-08-06T00:00:00.000Z',
  DeliveryType: 'E',
  Deposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '37.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'VA',
      },
    ],
  },
  Division: 'SE',
  EmailAddress: 'jamiamorrow@gmail.com',
  EstimatedDeliveryTimeFrame: '',
  FinDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '600.00',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'RGE',
      },
    ],
  },
  FinanceAccountNumber: '6019191263030424',
  FinancePlan: '90015-0121',
  FraudFlag: '',
  GCDeposits: {
    Deposit: [
      {
        AuthorizationCode: '004547',
        AuthorizedAmount: '1.95',
        AuthorizedDate: '2019-08-04T00:00:00.000Z',
        CashCode: 'GC',
      },
    ],
  },
  I95ind: false,
  LineItems: {
    LineItem: [
      {
        ChildItems: {
          LineItem: [
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36111639',
              UnitPrice: '193.24',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
            {
              Quantity: 1,
              SelectedMonogramFee: 0,
              Sku: '36211631',
              UnitPrice: '250.76',
              UnitProtectionPrice: '0',
              UnitProtectionType: 'N',
            },
          ],
        },
        PackagePrice: '444',
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '3611163P',
        UnitPrice: '0',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Louie Gray Twin Loft Bed with Desk and Bookcase',
      },
      {
        ChildItems: null,
        Quantity: 1,
        SelectedMonogramFee: 0,
        Sku: '38096613',
        UnitPrice: '29',
        UnitProtectionPrice: '0',
        UnitProtectionType: 'N',
        DisplayName: 'Mattie Gray Chevron Bean Bag Chair',
      },
    ],
  },
  OrderDate: '2019-08-04T23:17:03.000Z',
  OrderMessage: 'OK',
  OrderNumber: '14000136I',
  OrderTotal: '638.95',
  PickUpData: {
    Phone: '',
    PickupAddress: { Address1: '', Address2: '', City: '', State: '', Zip: '' },
    PickupHours: { string: null },
    StoreName: '',
    StoreNumber: '980',
  },
  RemoteDeliveryDate: '0001-01-01T00:00:00.000Z',
  RemoteEstimatedDeliveryTimeFrame: '',
  ShipDate: '',
  Status: 'Open',
  StoreNumber: '980',
  TaxAmount: '45.96',
  TaxAreaId: '111211811',
  Active: true,
  DispatchTrack: 'https://fgo-rtg.dispatchtrack.com/track_order/f0d44a2/SE14000136I',
}

export const orderStatusList = [orderStatus]
