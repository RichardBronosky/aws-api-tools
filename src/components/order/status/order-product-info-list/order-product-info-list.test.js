import React from 'react'
import renderer from 'react-test-renderer'
import OrderProductInfoList from '../order-product-info-list-item'
import { orderStatus } from '@mocks/orderStatusDataMocks'

describe('OrderProductInfoList', () => {
  const testProps = {
    order: orderStatus,
  }

  it('renders correctly with default props', () => {
    const tree = renderer.create(<OrderProductInfoList { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when orderStatus does not exist', () => {
    const tree = renderer.create(<OrderProductInfoList order={ null } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when orderStatus does not have line items', () => {
    const tree = renderer.create(<OrderProductInfoList order={ { ...orderStatus, LineItems: null } } />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
