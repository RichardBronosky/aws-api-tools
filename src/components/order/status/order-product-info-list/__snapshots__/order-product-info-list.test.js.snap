// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderProductInfoList renders correctly when orderStatus does not exist 1`] = `null`;

exports[`OrderProductInfoList renders correctly when orderStatus does not have line items 1`] = `null`;

exports[`OrderProductInfoList renders correctly with default props 1`] = `null`;
